### PR TITLE
feature(Plugins/Premium Plugins/a11ychecker): Update docs of a11checker

### PR DESCRIPTION
**Pull request by hamza0867**
Related Ticket: TINY-8595

Description of Changes:
This PR updates the docs of the Plugins/Premium Plugins/a11ychecker page. It assumes The PR #2280 is going to merge first, so there might be broken links or missing ids because they are done in the another 

Pre-checks:
- [x] Branch prefixed with `feature/` or `hotfix/`
- [x] `_data/nav.yml` has been updated (if applicable)
- [x] Files has been included where required (if applicable)
- [x] Files removed have been deleted, not just excluded from the build (if applicable)
- [x] (New product features only) Release Note added

Review:
- [x] Documentation Team Lead has reviewed
- [x] Product Manager has reviewed

### CODE REVIEW

File: modules/ROOT/pages/plugins/premium/a11ychecker.adoc [hunk 1]
```diff
@@ -28,7 +28,7 @@ tinymce.init({
   toolbar: 'a11ycheck'
 });
 ----
-
+[[accessiblityrules]]
```
Comment (lnewson) [inline modules/ROOT/pages/plugins/premium/a11ychecker.adoc:31]:
```suggestion

[[accessiblityrules]]
```

File: modules/ROOT/partials/configuration/a11ychecker_filter_issue.adoc [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 The `a11ychecker_filter_issue` option allows Accessibility Checker issues to be removed from the Accessibility Checker report using a callback function. This option is a flexible alternative to the `a11ychecker_ignored_rules` option. This option can remove issues from the results shown in the dialog and the `getReport` API.
 
-The function will be passed the results of the <<getreport,`getReport()` API>>. To filter an issue from the Accessibility Checker report, the callback needs to return `false` for that particular issue.
+The function will be passed the results of the xref:getreport[`getReport()` API]. To filter an issue from the Accessibility Checker report, the callback needs to return `false` for that particular issue.
```
Comment (lnewson) [inline modules/ROOT/partials/configuration/a11ychecker_filter_issue.adoc:5]:
```suggestion
The function will be passed the results of the xref:plugins/premium/a11ychecker.adoc#getreport[`getReport()` API]. To filter an issue from the Accessibility Checker report, the callback needs to return `false` for that particular issue.
```
This id doesn't appear to be on this page and instead it looks like it's meant to link back to the main A11ychecker plugin page.

File: modules/ROOT/partials/misc/plugin-toolbar-button-id-boilerplate.adoc [hunk 3]
```diff
@@ -1,28 +1,29 @@
-{% if page.name == "available-toolbar-buttons.md" %}
+ifeval::["{docname}" == "advanced/available-toolbar-buttons"]
```
Comment (lnewson) [inline modules/ROOT/partials/misc/plugin-toolbar-button-id-boilerplate.adoc:1]: We really need to get those first 2 PRs merged as Lorenzo also had to fix these pages... I had mentioned it this morning in hopes to get the approvals needed, but I'll try again tomorrow sorry.

**Review by lnewson — APPROVED** [review]

File: modules/ROOT/pages/plugins/premium/a11ychecker.adoc (hunk 1 above)
Comment (EkimChau) [reply modules/ROOT/pages/plugins/premium/a11ychecker.adoc:31]: The spelling for 'accessibility' needs to be updated on link and anchor
suggest:
[[accessibilityrules]]

**Review by EkimChau — APPROVED** [review]
Fix minor type :)

**Comment by hamza0867** [comment]
A lot of files changed because I rebased this current branch from the latest release/TINYMCE_5.10. It looks good to me, especially the parts that were fixed in other PRs.

File: modules/ROOT/partials/misc/plugin-toolbar-button-id-boilerplate.adoc (hunk 3 above)
Comment (hamza0867) [reply modules/ROOT/partials/misc/plugin-toolbar-button-id-boilerplate.adoc:1]: After rebasing off of the latest release/TinyMCE_5.10 things look good to me

File: modules/ROOT/pages/plugins/premium/a11ychecker.adoc [hunk 4]
```diff
@@ -269,15 +255,15 @@ NOTE: The table `summary` attribute was deprecated in HTML 5, both the *what* an
...
+* *xref:a11ychecker_html_version[HTML version]*: HTML4 and HTML5
 * *WCAG 2.1 specification*: https://www.w3.org/WAI/WCAG21/Techniques/html/H63.html[H63 - Using the scope attribute to associate header cells and data cells in data tables].
 
 == Configuration Options
 
 The following configuration options affect the behavior of the {pluginname} plugin.
 
 :includedSection: a11yPlugin
-#include::partial$configuration/a11y_advanced_options.adoc[]
+include::partial$configuration/a11y_advanced_options.adoc[]
```
Comment (lnewson) [inline modules/ROOT/pages/plugins/premium/a11ychecker.adoc:266]:
```suggestion
include::partial$configuration/a11y_advanced_options.adoc[leveloffset=+1]
```
I missed this when reviewing the other day sorry, but the nav/heading levels isn't correct so we need to adjust this via the include:
image

File: modules/ROOT/partials/configuration/a11ychecker_filter_issue.adoc [hunk 5]
```diff
@@ -2,7 +2,7 @@
```
Comment (lnewson) [inline modules/ROOT/partials/configuration/a11ychecker_filter_issue.adoc:2]: Sorry I missed this as well, but this needs an id added.

- [ ] Add ids to the options for a11ychecker

Comment (hamza0867) [reply modules/ROOT/partials/configuration/a11ychecker_filter_issue.adoc:3]: Done

**Pull request merged**